File: 2285-maximum-total-importance-of-roads/2285-maximum-total-importance-of-roads.java
import java.util.*;

class Solution {
    public long maximumImportance(int n, int[][] roads) {
        HashMap<Integer, Integer> mp = new HashMap<>();        
        for (int i = 0; i < roads.length; i++) {
            int e1 = roads[i][0];
            int e2 = roads[i][1];
            mp.put(e1, mp.getOrDefault(e1, 0) + 1);
            mp.put(e2, mp.getOrDefault(e2, 0) + 1);
        }
        
        ArrayList<Integer> nums = new ArrayList<>(mp.values());
        Collections.sort(nums, Collections.reverseOrder());
        
        long sum = 0;
        
        for (int i = 0; i < nums.size(); i++) {
            sum += (long)nums.get(i) * n--;
        }
        
        return sum;
    }
}

File: 2285-maximum-total-importance-of-roads/2285-maximum-total-importance-of-roads.cpp
using namespace std;
class Solution {
public:
    long long maximumImportance(int n, vector<vector<int>>& roads) {
        unordered_map<long, long> mp;
        for(int i = 0; i <roads.size(); i++) {
            mp[roads[i][0]]++;
            mp[roads[i][1]]++;
        }
        vector<long>vec;
        for(auto& p : mp) {
            vec.push_back(p.second);
        }
        sort(vec.begin(), vec.end(), greater<long>());
        long long sum = 0;
        for(int i = 0; i < vec.size(); i++) {
            sum += (long long)vec[i] * n--;
        }
        return sum;
    }
};
